scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,260 @@ output:
                        other_fields = "diagnosis_group") +
+  # split plots by diagnosis group 
   facet_wrap(~ diagnosis_group)
 ```
-We can also look at the distribution of cell types across each sample.
 
-```{r}
+As expected, we see that the visual clusters of cell types are being separated, most likely due to different RMS subtypes.
+
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
+
+```{r celltype barplot}
+# filter coldata data.frame to only include tumor cells
 tumor_cells_df <- coldata_df %>%
   dplyr::filter(stringr::str_detect(celltype_broad, "Tumor"))
 
+# create a stacked barplot
 ggplot(tumor_cells_df, aes(x = sample, fill = celltype_broad)) + 
-    geom_bar(color = "black", size = 0.2) +
+    geom_bar(position = "fill", color = "black", size = 0.2) +
     labs(
-      x = "Sample",
+      x = "Library",
       y = "Number of cells", 
       fill = "Cell type"
     ) +
   theme_bw() +
-  theme(axis.text.x = element_text(angle = 90))+
-  facet_wrap(~diagnosis_group, scales = "free_x")
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
+  facet_grid(cols = vars(diagnosis_group), 
+             scales = "free_x",
+             space = "free_x")
 ```
 
-*Need to explain the difference between tumor/tumor_myoblast, etc*
-These plots show that ARMS and ERMS share a lot of the same cell types, but perhaps there are differences in the cells between subdiagnoses.
-We also see that 6 of the libraries show a mixture of tumor cells belonging to the mesoderm, myoblast, and myocyte classes.
-3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified and the remaining library appears to not have been assigned any cell types, as they are all labeled with `NA`.
-We will not be using these 3 libraries.
+Similar to the UMAP, this plot shows that ARMS and ERMS share a lot of the same cell types.
 
-```{r}
-# define libraries to keep
+We also see that only 6 of these libraries have tumor cells that have been further classified into mesoderm, myoblast, and myocyte. 
+3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified, and the remaining library appears to not have been assigned any cell types and is not present in our plot, as they are all labeled with `NA`.
+We will continue our analysis only using the 6 libraries with fully classified cell types, removing the other 3 before we proceed with differential expression.
+
+The reason we want to pare down our list of samples to consider is that we want to ensure that the cell types (or subpopulations) that we are interested in are present in all samples included in our DE analysis. 
+We want to remove any samples that do not contain our cell population(s) of interest as they have no counts to contribute to the DE analysis.
```

Comment:
There's a bit of nuance here - those 3 libraries probably do have those (or at least some) of those cell types, but we don't know which ones those are because of classification issues.

Reply:
This is true, but I don't think I want to get too far into the weeds of the classifications here.